cellx/tools/dataset.py

```diff
@@ -160,3 +160,49 @@ def build_dataset(files: Union[List[os.PathLike], os.PathLike], **kwargs):
     dataset = tf.data.TFRecordDataset(files)
     dataset = dataset.map(lambda x: parse_tfrecord(x, **kwargs), num_parallel_calls=8)
     return dataset
+
+
+def count_images_in_dataset(
+    files: Union[List[os.PathLike], os.PathLike],
+) -> int:
+    """Parse a TF Dataset by file to count the number of images it contains.
+
+    Parameters
+    ----------
+    files : str, list[str]
+        The list of TFRecord files to use for the dataset.
+
+    Returns
+    -------
+    num_images : int
+        The number of images in the TF dataset.
+    """
+
+    dataset = build_dataset(files)
```

Comment:
Actually, now that I think about it - do we actually need to build the dataset here, as we don't need to read the entries. Perhaps we could change this to:

```suggestion
    dataset = tf.data.TFRecordDataset(files)
```

Which may lead to better performance?